src/utils/definition-helpers.ts

```diff
@@ -280,7 +280,7 @@ export function getAddonPathsForComponentTemplates(root: string, maybeComponentN
       .map((pathArr: string[]): string => {
         return path.join(...pathArr.filter((part: any) => !!part));
       })
-      .filter(fs.existsSync);
+      .filter((filePath) => (addonName ? filePath.includes(addonName) && fs.existsSync(filePath) : fs.existsSync(filePath)));
```

Comment:
we should not do it, because addonName != part of addon path, path may be `lib/addon-name-one`, but in package.json, name will be "my-addon", this function should be "dummy", we have top-level logic to handle it